algebraic/src/witness/witness_calculator.rs

```diff
@@ -52,19 +51,14 @@ fn to_array32(s: &BigInt, size: usize) -> Vec<u32> {
 }
 
 impl WitnessCalculator {
-    pub fn new(path: impl AsRef<std::path::Path>) -> Result<Self> {
-        Self::from_file(path)
-    }
-
-    pub fn from_file(path: impl AsRef<std::path::Path>) -> Result<Self> {
-        let store = Store::default();
+    pub fn from_file(path: impl AsRef<std::path::Path>) -> Result<(Store, Self)> {
+        let mut store = Store::default();
```

Comment:
We can keep the store as a member variable of WitnessCalculator, and avoid passing this variable for it's every public funciton.

Reply:
Done